### DISCUSSION TITLE
Support of Maven --projects --also-make

### DISCUSSION THREAD

**Discussion by pgaufillet** [eclipse-tycho/tycho#1017] [Q&A] [answered]
Hi,

when trying to partially build a project, maven provides a few advanced options like --projects (comma separated list of modules to build) and --also-make (also build modules required for the list of projects) which are very handy. Are these options supported by Tycho? It seems that --also-make in particular doesn't have any effect (the build fails during dependencies resolution).

**Comment by laeubi** [comment, marked as answer]
Tycho support this from version 2.7.0, but you need to enable the build extension:
https://github.com/eclipse/tycho/blob/master/RELEASE_NOTES.md#tycho-pomless-will-become-a-tycho-core-extension
https://github.com/eclipse/tycho/blob/master/RELEASE_NOTES.md#tycho-specific-maven-graphbuilder-to-support---also-make--am-and---also-make-dependents--amd

**Comment by pgaufillet** [comment]
Thank you @laeubi: I missed it in the release notes.  Problem solved!

To complete (because this explanation is elsewhere in the changelog), the required extension can be activated by adding a `.mvn/extensions.xml` file in the `pom.xml` parent directory with the following content:
```
<?xml version="1.0" encoding="utf-8" ?>
<extensions>
    <extension>
        <groupId>org.eclipse.tycho</groupId>
        <artifactId>tycho-build</artifactId>
        <version>2.7.3</version>
    </extension>
</extensions>
```

**Reply by laeubi** [reply]
@pgaufillet if you think the release noted could be enhanced, simply hit the edit button on GH and suggest a change! 
Release notes are mostly written by the developers and thus not always user friendly :-)

**Comment by pgaufillet** [comment]
Ok. 
By the way, I am facing an unexpected behavior during the partial build of a product:
```
mvn -pf c.a.n.s.product -am clean package
[...]
[INFO] --- tycho-p2-publisher-plugin:2.7.3:publish-products (default-publish-products) @ c.a.n.s.product ---
[ERROR] Cannot resolve dependencies of product n-s.product:
[ERROR]   eclipse-plugin artifact with ID "c.a.n.api" and version matching "0.0.0" was not found in the target platform
[...]
```

And it is repeated for each bundle that is a module of the parent pom.xml.
This behavior doesn't appear when building usual bundles.

**Reply by laeubi** [reply]
It is possible that this case if not yet fully supported, the best way would be to enhance the integration test here:

- https://github.com/eclipse/tycho/tree/master/tycho-its/projects/reactor.makeBehaviour
- https://github.com/eclipse/tycho/blob/master/tycho-its/src/test/java/org/eclipse/tycho/test/reactor/makeBehaviour/MavenReactorMakeOptionsTest.java

to show the issue so we can improve here.

**Reply by joeshannon** [reply]
If I remember correctly, I think [this method](https://github.com/eclipse/tycho/blob/b9c5d7304593a05795787a9f072300324bb6c066/tycho-build/src/main/java/org/eclipse/tycho/build/TychoGraphBuilder.java#L264) would need to be extended to support product IUs.

**Reply by laeubi** [reply]
This is currently under work here: https://github.com/eclipse/tycho/pull/1050
